### PR TITLE
Make copy when deleting instead of messing up range operations

### PR DESCRIPTION
**Pull request by eparis**
We would walk the labels and comments lists and could then delete from
them. Since we used the form `a = append(a[:i], a[i+1:...])` we were
actually changing the list as it was being used in a `range a`
statement. This could cause the comment deleter and the LGTM label
checker to go nuts.

### CODE REVIEW

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -654,7 +654,12 @@ func (obj *MungeObject) RemoveLabel(label string) error {
 		}
 	}
 	if which != -1 {
-		obj.Issue.Labels = append(obj.Issue.Labels[:which], obj.Issue.Labels[which+1:]...)
+		// We do this crazy delete since users might be iterating over `range obj.Issue.Labels`
+		// Make a completly new copy and leave their ranging alone.
```
Comment (knobunc) [inline mungegithub/github/github.go]: completely

File: mungegithub/github/github.go [hunk 2]
```diff
@@ -1345,13 +1350,22 @@ func (obj *MungeObject) DeleteComment(comment *github.IssueComment) error {
...
 		if c.ID == nil || *c.ID != *comment.ID {
 			continue
 		}
-		obj.comments = append(obj.comments[:i], obj.comments[i+1:]...)
-		break
+		which = i
+	}
+	if which != -1 {
+		// We do this crazy delete since users might be iterating over `range obj.comments`
+		// Make a completly new copy and leave their ranging alone.
```
Comment (knobunc) [inline mungegithub/github/github.go]: completely

**Comment by knobunc** [comment]
LGTM

**Comment by eparis** [comment]
Thank you @knobunc @jayunit100

**Pull request merged**